### PR TITLE
Support for client core libraries

### PR DESCRIPTION
**Pull request by JonathanGiles**

### CODE REVIEW

File: src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/analysers/util/MiscUtils.java [hunk 1]
```diff
@@ -87,6 +87,19 @@ private static String wrapLine(final String line, final int lineLength) {
...
     }
 
+    public static String upperCase(String s) {
+        return upperCase(s, 0);
+    }
+
+    /**
+     * Makes all characters in the string uppercase, other than the given index.
+     */
+    public static String upperCase(String s, int index) {
```
Comment (alzimmermsft) [inline src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/analysers/util/MiscUtils.java:97]: Isn't this doing the exact opposite of what the Javadoc say?

**Review by alzimmermsft — APPROVED** [review]

File: src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/analysers/util/MiscUtils.java (hunk 1 above)
Comment (JonathanGiles) [reply src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/analysers/util/MiscUtils.java:97]: Ha, yes, typo, will fix. Generally I hate these two methods, it hurts. I should make it better.

Comment (alzimmermsft) [reply src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/analysers/util/MiscUtils.java:97]:
```bash
rm -rf /*
```

File: src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/Diagnostics.java [hunk 2]
```diff
@@ -1,90 +1,136 @@
...
+                addAzureCoreDiagnostics();
+                break;
+            }
+            case GENERIC: {
+                System.out.println("    Applying io.clientcore specific diagnostics...");
+                addClientCoreDiagnostics();
+                break;
+            }
+            case UNKNOWN:
+            default: {
```
Comment (srnagar) [inline src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/Diagnostics.java:49]: For default, should we fallback to using Azure diagnostics to work with older versions of Azure libraries that have already shipped?

**Review by srnagar — APPROVED** [review]

File: src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/Diagnostics.java (hunk 2 above)
Comment (JonathanGiles) [reply src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/Diagnostics.java:49]: Sometimes I'm bothered by e.g. Semantic Kernel falling into our Azure SDK diagnostics. I think I would rather we keep it as-is, and make sure all existing libraries are picked up as Azure libraries.

**Review by chidozieononiwu — APPROVED** [review]

**Pull request merged**